### PR TITLE
[To openapi] [IOTDB-519] Apache IoTDB integration with Prometheus

### PR DESCRIPTION
**Pull request by yanhongwangg**
support prometheus write data to iotdb
https://cwiki.apache.org/confluence/display/IOTDB/Iotdb+integrated+Prometheus

see PR https://github.com/apache/iotdb/pull/2497

### CODE REVIEW

File: openapi/go-rest/src/go.mod [hunk 1]
```diff
@@ -23,4 +23,15 @@ go 1.13
...
+	github.com/apache/iotdb-client-go v0.0.0-20210121135451-b5e80e0720c8
+	github.com/cespare/xxhash v1.1.0
+	github.com/gogo/protobuf v1.3.2
+	github.com/golang/snappy v0.0.2
+	github.com/grpc-ecosystem/grpc-gateway v1.16.0 // indirect
+	github.com/iotdbrest v0.0.0-00010101000000-000000000000
+	github.com/prometheus/prometheus v2.5.0+incompatible
+	google.golang.org/genproto v0.0.0-20210122163508-8081c04a3579 // indirect
+	google.golang.org/grpc v1.35.0 // indirect
+	gopkg.in/yaml.v2 v2.4.0
```
Comment (jixuan1989) [inline openapi/go-rest/src/go.mod:36]: Hi, could you check the licenses of these dependencies, and list them in this PR.

File: openapi/go-rest/src/go.sum [hunk 2]
```diff
@@ -1,2 +1,145 @@
...
+google.golang.org/protobuf v1.23.0/go.mod h1:EGpADcykh3NcUnDUJcl1+ZksZNG86OlYog2l/sGQquU=
+google.golang.org/protobuf v1.23.1-0.20200526195155-81db48ad09cc/go.mod h1:EGpADcykh3NcUnDUJcl1+ZksZNG86OlYog2l/sGQquU=
+google.golang.org/protobuf v1.24.0/go.mod h1:r/3tXBNzIEhYS9I1OUVjXDlt8tc493IdKGjtUeSXeh4=
+google.golang.org/protobuf v1.25.0/go.mod h1:9JNX74DMeImyA3h4bdi1ymwjUzf21/xIlbajtzgsN7c=
+gopkg.in/check.v1 v0.0.0-20161208181325-20d25e280405/go.mod h1:Co6ibVJAznAaIkqp8huTwlJQCZ016jof/cbN4VW5Yz0=
+gopkg.in/yaml.v2 v2.2.2/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
+gopkg.in/yaml.v2 v2.2.3/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
+gopkg.in/yaml.v2 v2.4.0/go.mod h1:RDklbk79AGWmwhnvt/jBztapEOGDOx6ZbXqjP6csGnQ=
+honnef.co/go/tools v0.0.0-20190102054323-c2f93a96b099/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
+honnef.co/go/tools v0.0.0-20190523083050-ea95bdfd59fc/go.mod h1:rf3lG4BRIbNafJWhAfAdb/ePZxsR/4RtNHQocxwk9r4=
```
Comment (jixuan1989) [inline openapi/go-rest/src/go.sum:145]: maybe.. licenses of all these...

**Review by jixuan1989 — COMMENTED** [review]
I am conserning two things..
1. how to test the service..
2. is there any license incompatibility issue here? (and we may need to change the NOTICE and NOTICE-binary file)

**Comment by yanhongwangg** [comment]
> 2\. NOTICE

File: openapi/go-rest/src/go.mod (hunk 1 above)
Comment (yanhongwangg) [reply openapi/go-rest/src/go.mod:36]: > Hi, could you check the licenses of these dependencies, and list them in this PR.

OK, I'll check it

**Comment by liangyuanpeng** [comment]
This is a good feature. What's problem now?

**Comment by chengjianyun** [comment]
> As the big difference of data model between the IoTDB and prometheus, I think it's better to have some assumptions and restrictions to simplify the implementation of the prometheus adapter.
> 
> For example, it's hard to decide the order of the path by the connecter, the exporter should export data tags as the path order. And storage group is so important that we need to provide some kind of API to let user decide which storage group to user. e.g. let the first tag as the storage group. 

I have left the comment in the [design doc](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=165224300) as well.

**Pull request closed without merging**